backend/app/adapter/graphql/resolver/authquery.go

```diff
@@ -35,9 +37,17 @@ func (v AuthQuery) URL(args *URLArgs) (*URL, error) {
 	return &URL{url: u}, nil
 }
 
-func newAuthQuery(user *entity.User, urlRetriever url.Retriever) AuthQuery {
+// ChangeLog retrieves full ChangeLog from persistent storage
+func (v AuthQuery) ChangeLog() (ChangeLog, error) {
+	changeLog, err := v.changeLog.GetChangeLog()
+	currentTime := time.Now()
```

Comment:
It is better to use our own `timer`. For example for tests purposes

Reply:
Yea, you are right. But we will be fetching lastViewedAt from storage in the next PR. This is just a temporary value for lastViewedAt..i think we should just set it to null for now instead of setting it to a temporary value
`return newChangeLog(changeLog, nil), err`

Reply:
ok, we will use mdtimer.NewTimer() for now

Reply:
@kapillamba4 Aha, we actually mean `fw.timer`, which is an interface. `mdtimer.Timer` is concrete implementation. We can't replace it on the fly.

```golang
type AuthQuery struct {
	user         entity.User
        timer       fw.timer
	changeLog    changelog.ChangeLog
	urlRetriever url.Retriever
}

func (a AuthQuery) ChangeLog() (ChangeLog, error) {
	changeLog, err := v.changeLog.GetChangeLog()
	currentTime := a. timer.Now()
	return newChangeLog(changeLog, currentTime), err
}
```